osgp/protocol-adapter-mqtt/osgp-protocol-adapter-mqtt/src/main/java/org/opensmartgridplatform/adapter/protocol/mqtt/application/services/MqttClient.java

```diff
@@ -99,4 +107,19 @@ private static void onDisconnect(final Void nothing, final Throwable throwable)
       LOGGER.info("MQTT connection disconnected.");
     }
   }
+
+  private static String getClientId() {
+    String value;
+    InetAddress inetAddress;
+    try {
+      inetAddress = InetAddress.getLocalHost();
+      value = inetAddress.getHostName() + '-' + inetAddress.getHostAddress();
```

Comment:
Asking for a non-clean-session with a random (kubernetes) hostname can result in loss of data. Maybe add a `clientName` config property which, if present, is used instead of the hostname?

Reply:
Added `mqtt.default.client.id` / `defaultClientId` property.